radixdlt/src/main/java/com/radixdlt/DispatcherModule.java

```diff
@@ -0,0 +1,116 @@
+import com.google.inject.Provides;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.Multibinder;
+import com.radixdlt.consensus.Timeout;
+import com.radixdlt.consensus.bft.BFTCommittedUpdate;
+import com.radixdlt.consensus.bft.BFTUpdate;
+import com.radixdlt.consensus.epoch.EpochView;
+import com.radixdlt.consensus.sync.LocalGetVerticesRequest;
+import com.radixdlt.environment.Environment;
+import com.radixdlt.environment.EventDispatcher;
+import com.radixdlt.environment.EventProcessor;
+import com.radixdlt.environment.ProcessOnDispatch;
+import com.radixdlt.environment.ScheduledEventDispatcher;
+import com.radixdlt.sync.LocalSyncRequest;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Manages dispatching of internal events to a given environment
+ * TODO: Move all other events into this module
+ */
+public class DispatcherModule extends AbstractModule {
+	private static final Logger logger = LogManager.getLogger();
+	@Override
+	public void configure() {
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<LocalSyncRequest>>() { }, ProcessOnDispatch.class);
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<LocalSyncRequest>>() { });
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<Timeout>>() { }, ProcessOnDispatch.class);
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<Timeout>>() { });
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<EpochView>>() { }, ProcessOnDispatch.class);
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<EpochView>>() { });
+		Multibinder.newSetBinder(binder(), new TypeLiteral<EventProcessor<BFTCommittedUpdate>>() { });
+	}
+
+	@Provides
+	private EventDispatcher<LocalSyncRequest> localSyncRequestEventDispatcher(
+		@ProcessOnDispatch Set<EventProcessor<LocalSyncRequest>> syncProcessors,
+		Environment environment
+	) {
+		EventDispatcher<LocalSyncRequest> envDispatcher = environment.getDispatcher(LocalSyncRequest.class);
+		return req -> {
+			Class<?> callingClass = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE).getCallerClass();
+			logger.info("LOCAL_SYNC_REQUEST dispatched by {}", callingClass);
+			syncProcessors.forEach(e -> e.process(req));
+			envDispatcher.dispatch(req);
+		};
+	}
+
+	@Provides
+	private ScheduledEventDispatcher<LocalGetVerticesRequest> localGetVerticesRequestRemoteEventDispatcher(Environment environment) {
+		return environment.getScheduledDispatcher(LocalGetVerticesRequest.class);
+	}
+
+	@Provides
+	private EventDispatcher<BFTUpdate> viewEventDispatcher(
+		Environment environment
+	) {
+		return environment.getDispatcher(BFTUpdate.class);
+	}
+
+	@Provides
+	private EventDispatcher<EpochView> viewEventDispatcher(
+		@ProcessOnDispatch Set<EventProcessor<EpochView>> processors,
+		Set<EventProcessor<EpochView>> asyncProcessors,
```

Comment:
These are required so that if there are no asyncProcessors then we skip dispatch the event into the environment. This is currently important because we've got deterministic tests which check for specific events in order and we dont want to include unused events in that queue.